common/types.go

```diff
@@ -275,3 +276,12 @@ type SubscriptionEvent interface {
 	ObjectName() (string, error)
 	Workspace() (string, error)
 }
+
+// WebhookVerificationParameters is a struct that contains the parameters required to verify a webhook.
+type WebhookVerificationParameters struct {
```

Comment:
Can we redefine this to have `http.Request` along with the client secret instead? That way, if another provider needs a different kind of verification (with some additional fields) for the webhook, we don't have to change this type, and we also get all the handy methods + libraries on top of http.Request

Reply:
http.Request is not meant to be marshaled due to some fields e.g) `Body` is not marshallable because it is a function interface. 

Really, what we need is all here, headers, body, URL, method. Also the most we will be using are getting headers, then json marshal body. So having the entire HTTP library will be overkill.